src/account/kernel/api/installModule.ts

```diff
@@ -0,0 +1,107 @@
+import { Account, Execution } from '../../types'
+import {
+  PublicClient,
+  encodeAbiParameters,
+  encodeFunctionData,
+  parseAbi,
+  slice,
+} from 'viem'
+import { isModuleInstalled } from './isModuleInstalled'
+import { accountAbi } from '../constants/abis'
+import { KernelModule, kernelModuleTypeIds } from '../types'
+
+export const installModule = ({
+  client,
+  account,
+  module,
+}: {
+  client: PublicClient
+  account: Account
+  module: KernelModule
+}): Promise<Execution[]> => {
+  switch (module.type) {
+    case 'validator':
+    case 'executor':
+    case 'hook':
+    case 'policy':
+    case 'signer':
+      return _installModule({ client, account, module })
+    case 'fallback':
+      return installFallback({ client, account, module })
+    default:
+      throw new Error(`Unknown module type ${module.type}`)
+  }
+}
+
+const _installModule = async ({
+  client,
+  account,
+  module,
+}: {
+  client: PublicClient
+  account: Account
+  module: KernelModule
+}) => {
+  const executions: Execution[] = []
+  const isInstalled = await isModuleInstalled({ client, account, module })
+
+  if (!isInstalled) {
+    executions.push({
+      target: account.address,
+      value: BigInt(0),
+      callData: encodeFunctionData({
+        functionName: 'installModule',
+        abi: parseAbi(accountAbi),
+        args: [
+          BigInt(kernelModuleTypeIds[module.type]),
+          module.module,
+          module.data || '0x',
+        ],
+      }),
+    })
+  }
+  return executions
```

Comment:
one thing that we should think about is changing this flow; currently, we return an array of executions with self calls to the account so the call flow is ep -> account.execute -> account.installModule but it would make more sense to have the flow be ep -> account.installModule